src/components/DataCollector/validateUserEventOptions.js

```diff
@@ -10,18 +10,39 @@ OF ANY KIND, either express or implied. See the License for the specific languag
 governing permissions and limitations under the License.
 */
 
+import { string, objectOf, boolean, arrayOf } from "../../utils/validation";
+
 /**
  * Verifies user provided event options.
  * @param {*} options The user event options to validate
- * @returns {Array} Array of warnings if the options are invalid
+ * @returns {Array} Array of errors if the options are invalid
  */
 export default options => {
-  const warnings = [];
-  const { xdm } = options;
-  if (!xdm && !options.data) {
-    warnings.push("No event xdm or event data specified.");
-  } else if (xdm && !xdm.eventType && !options.type) {
-    warnings.push("No type or xdm.eventType specified.");
+  const eventOptionsValidator = objectOf({
+    viewStart: boolean(),
+    type: string().nonEmpty(),
+    xdm: objectOf({
+      eventType: string().nonEmpty()
+    }).nonEmpty(),
+    data: objectOf({}).nonEmpty(),
+    scopes: arrayOf(
+      string()
+        .required()
+        .nonEmpty()
+    ).nonEmpty()
+  })
+    .required()
+    .nonEmpty();
+  const errors = [];
+  try {
+    const { type, xdm, data, scopes } = eventOptionsValidator(options);
+    if (!xdm && !data && !scopes) {
+      errors.push("No event xdm, data, or scopes specified.");
+    } else if (xdm && !xdm.eventType && !type) {
+      errors.push("No type or xdm.eventType specified.");
```

Comment:
A few reasons, we're treating option verification as errors in personalization and validation logic throws and labels validation problems as errors. It will also force users to provide proper event syntax.

Reply:
Is Justin aware of this change? Last time we talked, he wanted to encourage but not require `eventType`.

Reply:
Correct. We should not require this. It should be a warning at most.